packages/docusaurus-plugin-content-docs/src/metadata.ts

```diff
@@ -98,13 +98,18 @@ export default async function processMetadata({
   // Default base id is the file name.
   const baseID: string =
     frontMatter.id || path.basename(source, path.extname(source));
-
   if (baseID.includes('/')) {
     throw new Error('Document id cannot include "/".');
   }
 
-  // Append subdirectory as part of id.
+  const baseSlug: string = frontMatter.slug || baseID;
+  if (baseSlug.includes('/')) {
+    throw new Error('Document slug cannot include "/".');
+  }
+
+  // Append subdirectory as part of id/slug.
   const id = dirName !== '.' ? `${dirName}/${baseID}` : baseID;
+  const slug = dirName !== '.' ? `${dirName}/${baseSlug}` : baseSlug;
```

Comment:
For anyone reviewing, it might be a good idea to sanitize the slug.

I didn't do it in this PR (apart from forbidden `/`, because it was already possible to use special characters in the `id` frontmatter anyway, from which the slug is inferred.

Reply:
What other sanitization do you think we should do?

Reply:
We should probably ensure that the id/slug does not contain forbidden url path segment chars.

Currently I tried:

![image](https://user-images.githubusercontent.com/749374/82550104-4ed47900-9b5e-11ea-9e18-981b9735b175.png)


This does not lead to a build error, but leads to a runtime error:

![image](https://user-images.githubusercontent.com/749374/82550160-66136680-9b5e-11ea-83bb-8d10df001c07.png)


I think we should whether sanitize that url (ie replace/remove illegal pathname chars), or simply reject the id/slug at build time instead

Reply:
Ah I see. Yeah, this is definitely something we should consider in another PR. Good point.